### PR TITLE
Yarn 2 PnP support

### PR DESCRIPTION
**Pull request by alexkuc**
Great template! Works out-of-the-box and everything is awesome!

Though it seems this template is not compatible with [Yarn 2 PnP](https://yarnpkg.com/features/pnp).

I have updated `gulpfile.js` to provide support for both, Yarn 2 and Yarn 1 (classic).

I have also updated `.gitignore` to exclude Yarn 2 PnP cache but if this is inappropriate I can remove it.

Let me know what you think! 😺

### CODE REVIEW

**Comment by alexkuc** [comment]
My only concern is whether https://github.com/StartBootstrap/startbootstrap-clean-blog/blob/fd4997c6ee12042c68532a6455439e7b348a7fe7/gulpfile.js#L77 is affected (sass).

Importing works fine, as-in, the final `.css` is _not_ empty and contains the expected selectors.

File: gulpfile.js [hunk 1]
```diff
@@ -47,21 +47,32 @@ function clean() {
...
+function module_path(module) {
+  // Yarn 2 Pnp
+  if (process.versions.pnp) {
+    return (require('path')).dirname(require.resolve(`${module}/package.json`));
+  }
+  // Yarn 1 (classic)
+  if (!process.versions.pnp) {
+    return `./node_modules/${module}`;
+  }
+}
```
Comment (merceyz) [inline gulpfile.js:59]: This should use `require.resolve` no matter what the user is running under, hardcoding `node_modules` paths is asking for issues, if for example this was used in a monorepo the path would (probably) not exist

Comment (alexkuc) [reply gulpfile.js:59]: I agree. I will update PR once I have some free time.

**Comment by alexkuc** [comment]
@merceyz if this looks good and no further editing is required, I can squash commits into one

**Comment by merceyz** [comment]
Looks good :+1:

**Comment by alexkuc** [comment]
I'll squash the commits so you can merge the PR

**Comment by alexkuc** [comment]
Commits squashed into one, you can merge

**Comment by merceyz** [comment]
I have no power here

**Comment by alexkuc** [comment]
Oh, right. We need one of the contributors to review.

**Comment by davidtmiller** [comment]
Unfortunately we dropped gulp as a dependency and moved to raw scripts so this change will not be added in.

**Pull request closed without merging**